### ISSUE TITLE
Support voice input on macOS

### ISSUE THREAD

**Issue by nekowasabi** [wezterm/wezterm#4592]
### What Operating System(s) are you seeing this problem on?

macOS

### Which Wayland compositor or X11 Window manager(s) are you using?

No.

### WezTerm version

Nightly 20231117-151105-60065258

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

No, and I'll explain why below

### Describe the bug

As a result of trying to activate the MacOS voice input function in Wezterm, the voice input icon does not appear and no text is entered on the wezterm screen when I vocalize. I am able to use the standard MacOS Terminal and iTerm2.

### To Reproduce

1. Open Wezterm 2. Focus Cursor 3. Type F2(default mac os voice input shortcut key

### Configuration
local wezterm = require("wezterm")
local act = wezterm.action

return {
  use_ime = true,
  hide_tab_bar_if_only_one_tab = true,
  enable_tab_bar = true,
	disable_default_key_bindings = true,
  keys = {
    { key = 'v', mods = 'CMD',  action=wezterm.action{PasteFrom="Clipboard"}},
  },
}

### Expected Behavior

Show voice input icon.

**Comment by jackfranklin** [comment]
I ended up here after googling as I'd love to have this enabled, particularly given that it's quite fun to talk to AI tools rather than type!

https://github.com/wezterm/wezterm/issues/5180 sounds like a duplicate of this. It links to [this PR in iTerm2](github.com/gnachman/iTerm2/pull/456/files) where they added support (I think?). 

Do you have any sense of how hard this would be to support? Presumably we mostly "just" need to give it permission to hook into the OS dictation permissions system to allow it to accept the voice input. I would be happy to attempt to contribute here if possible.

**Comment by BarakXYZ** [comment]
It would be amazing to have this feature!

Edit:
For now, I've worked around this using this very nice tool called [hear](https://github.com/sveinbjornt/hear?tab=readme-ov-file).
It gives a CLI to the built-in Speech Recognition API Apple has. We can then automate dictation to places Mac won't support with this vibe-coded script (lol):
```bash
#!/bin/bash

PID_FILE="/tmp/hear-pid.txt"
OUTPUT_FILE="/tmp/dictation-output.txt"

# Check if hear is running (by checking if PID file exists and process is alive)
if [ -f "$PID_FILE" ] && kill -0 $(cat "$PID_FILE" 2>/dev/null) 2>/dev/null; then
    # Wait for the dictation to stabilize final output (feel free to play around with this)
    sleep 1.8
    
    # Kill the hear process using stored PID
    kill -TERM $(cat "$PID_FILE") 2>/dev/null
    sleep 0.1
    kill -KILL $(cat "$PID_FILE") 2>/dev/null  # Force kill if still running
    
    # Wait a moment for file to be written
    sleep 0.1
    
    # Copy only the last line (final result) to clipboard
    if [ -f "$OUTPUT_FILE" ]; then
        tail -n 1 "$OUTPUT_FILE" | tr -d '\n' | pbcopy
        echo "Copied to clipboard"
    fi
    
    # Clear the output file and remove PID file
    > "$OUTPUT_FILE"
    rm -f "$PID_FILE"
    
    # Close ALL Terminal windows and quit Terminal if no other windows
    osascript << 'EOF'
tell application "Terminal"
    close every window
    delay 0.5
    if (count of windows) = 0 then quit
end tell
EOF
    
    # Auto-paste to the frontmost app (not Terminal)
    osascript << 'EOF'
tell application "System Events"
    delay 0.2
    keystroke "v" using command down
end tell
EOF
    
else
    # Clear the output file before starting
    > "$OUTPUT_FILE"
    
    # Start hear in Terminal, minimize it immediately, and capture PID
    osascript << 'EOF'
tell application "Terminal"
    -- Create new window with a command that captures PID silently
    set newWindow to do script "sh -c 'hear -p > /tmp/dictation-output.txt & echo $! > /tmp/hear-pid.txt; wait'"
    
    -- Minimize the window immediately
    tell application "System Events"
        tell process "Terminal"
            perform action "AXMinimize" of button 2 of window 1
        end tell
    end tell
end tell
EOF
fi
```
You can then trigger this via something like [Karabiner](https://github.com/pqrs-org/Karabiner-Elements) and have this entire process feel more natural like the native dictation tool. You won't get real-time dictation preview, but it's still functional and pastes the final result well.

**Comment by rjhilgefort** [comment]
+1 for this feature. I've been using Claude Code and would appreciate being able to talk my prompts instead of typing them.

**Comment by johanbelin** [comment]
switched from terminal just got used to dictation, please add

**Comment by halidaee** [comment]
WezTerm misbehaves with Superwhisper as well due to a similar mechanism. (With superwhisper, the dictated text is pasted but the superwhisper window doesn't close afterwards; closing must be done manually unlike in iterm2 which works seamlessly.)

**Comment by Askath** [comment]
Also would love to have this implemented. I cannot use whisper or any other AI tools on my machine due to company policy but macos dictation is fine.

**Comment by secretisgratitude** [comment]
Confirming this is still present on the current stable release (20240203-110809-5046fc22, installed via Homebrew cask, which is also the latest tag on the releases page as of now). Same symptom: macOS dictation (tested both default Ctrl-double-tap and Command-double-tap shortcuts) does nothing in a WezTerm window — no mic icon, no text — while working correctly in the same session for Terminal.app and other apps on the same machine. The shortcut itself fires fine at the OS level; it's specifically not reaching WezTerm's window. Wanted to note it's not fixed as of the current release, in case anyone assumes this was resolved by #7536 (the selectedRange fix) — that doesn't appear to be in a tagged release yet.
